tuf/client/updater.py

```diff
@@ -643,6 +643,13 @@ def refresh(self, unsafely_update_root_if_necessary=True):
         logger.info('An expired Root metadata was loaded and must be updated.')
         raise
 
+    # If an exception is raised during the metadata update attempts, we will
+    # attempt to update root metadata once by recursing with a special argument
+    # to avoid further recursion. We use this bool and pull the recursion out
+    # of the except block so as to avoid unprintable nested NoWorkingMirrorError
```

Comment:
I'll try to reproduce the issue again. Resolved it by moving that out of the except block, but perhaps there's a better way.

Was this:

```
======================================================================
ERROR: test_with_tuf (__main__.TestIndefiniteFreezeAttack)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "test_indefinite_freeze_attack.py", line 284, in test_with_tuf
	self.repository_updater.refresh()
  File "/home/s/w/tuf/tuf/client/updater.py", line 699, in refresh
	self.refresh(unsafely_update_root_if_necessary=False)
  File "/home/s/w/tuf/tuf/client/updater.py", line 651, in refresh
	referenced_metadata='timestamp')
  File "/home/s/w/tuf/tuf/client/updater.py", line 1674, in _update_metadata_if_changed
	expected_versioninfo['version'], compression)
  File "/home/s/w/tuf/tuf/client/updater.py", line 1468, in _update_metadata
	compression_algorithm)
  File "/home/s/w/tuf/tuf/client/updater.py", line 1189, in _get_metadata_file
	raise tuf.NoWorkingMirrorError(file_mirror_errors)
tuf.NoWorkingMirrorError: <unprintable NoWorkingMirrorError object>

----------------------------------------------------------------------
Ran 2 tests in 16.959s

FAILED (errors=1)

```